### PR TITLE
[Breaking] Drop Enzyme support from Jest preset and make it opt-in

### PR DESCRIPTION
**Pull request by emmenko**
Ref #1829

### CODE REVIEW

File: .changeset/serious-actors-attend.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+### Enzyme 17
+
+1. Install the necessary dependencies:
+
+```
+yarn add \
+  @commercetools/enzyme-extensions \
+  @commercetools/jest-enzyme-matchers \
+  @wojtekmaj/enzyme-adapter-react-17 \
```
Comment (adnasa) [inline .changeset/serious-actors-attend.md:46]: worth mentioning real quick that this is not the official package. it's easy to miss!

**Review by tdeekens — APPROVED** [review]
Nice clean up! 💯

File: packages/jest-preset-mc-app/enzyme/create-jest-preset-for-enzyme.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  path.resolve(__dirname, relativePath);
+
+const createJestPresetForEnzyme = (version) => ({
+  ...defaultPreset,
+  snapshotSerializers: ['enzyme-to-json/serializer'],
+  setupFilesAfterEnv: [
+    resolveRelativePath('../setup-test-framework.js'),
+    resolveRelativePath(`./setup-test-framework-for-enzyme-${version}.js`),
+  ],
+});
```
Comment (adnasa) [inline packages/jest-preset-mc-app/enzyme/create-jest-preset-for-enzyme.js:14]: nice abstraction

**Review by adnasa — APPROVED** [review]
hammer geil

**Comment by adnasa** [comment]
to be clear, when we release this, we update the MC-FE to opt-in v16 adapter, right?
I don't mind doing it.

**Pull request merged**